benchmarks/src/main/scala/ai/h2o/sparkling/benchmarks/Runner.scala

```diff
@@ -0,0 +1,162 @@
+    // set it to the class name.
+    val appName = conf.get("spark.app.name", this.getClass.getSimpleName)
+    conf.setAppName(appName)
+
+    conf
+  }
+
+  def main(args: Array[String]): Unit = {
+    val settings = processArguments(args)
+    val datasetDetails = loadDatasetDetails()
+    val benchmarks = getBenchmarkClasses()
+    val algorithms = AlgorithmBenchmarkBase.supportedAlgorithms
+
+    val filteredDatasetDetails = filterCollection[DatasetDetails]("Dataset", settings.dataset, datasetDetails, _.name)
+    val filteredBenchmarks = filterCollection[Class[_]]("Benchmark", settings.benchmark, benchmarks, _.getSimpleName)
+    val filteredAlgorithms = filterCollection[H2OAlgorithm[_, _, _]]("Algorithm", settings.algorithm, algorithms, _.getClass.getSimpleName)
+
+    val batches = createBatches(filteredDatasetDetails, filteredBenchmarks, filteredAlgorithms)
+    batches.foreach(executeBatch)
+
+    hc.stop(stopSparkContext = true)
+  }
+
+  private def processArguments(args: Array[String]): Settings = {
+    require(args.length % 2 == 0, "Wrong arguments. Example: -b benchmarkName -d datasetName -a algorithmName")
+    val (keys, values) = args.zipWithIndex.partition{ case (_, idx) => idx % 2 == 1 }
+    val map = keys.map(_._1).zip(values.map(_._1)).toMap
+    Settings(map.get("-b"), map.get("-d"), map.get("-a"))
+  }
+
+  private def loadDatasetDetails(): Seq[DatasetDetails] = {
+    val url = getClass.getClassLoader.getResource(datasetDetailsFilePath)
+    implicit val formats = DefaultFormats
+    val reader = new FileReader(url.getPath)
+    try {
+      read[Seq[DatasetDetails]](reader)
+    } finally {
+      reader.close()
+    }
+  }
+
+  private def getBenchmarkClasses(): Seq[Class[_]] = {
+    val classLoader = Thread.currentThread().getContextClassLoader
+    val classPath = ClassPath.from(classLoader)
+    val packageName = this.getClass.getPackage.getName
+
+    def isBenchmark(clazz: Class[_]) = {
+      val isAbstract = Modifier.isAbstract(clazz.getModifiers)
+      val inheritsFromBenchmarkBase  = classOf[BenchmarkBase].isAssignableFrom(clazz)
+      !isAbstract && inheritsFromBenchmarkBase
+    }
+
+    val classes = classPath.getTopLevelClasses(packageName).asScala.map(_.load())
+    classes.filter(isBenchmark(_)).toSeq
+  }
+
+  private def filterCollection[T](
+      entity: String,
+      filter: Option[String],
+      collection: Seq[T],
+      nameGetter: T => String): Seq[T] = filter match {
+    case None => collection
+    case Some(name) =>
+      val result = collection.filter(nameGetter(_) == name)
+      require(result.length > 0, s"$entity '$name' does not exist!")
+      result
+  }
+
+  private def createBatches(
+      datasetDetails: Seq[DatasetDetails],
+      benchmarkClasses: Seq[Class[_]],
+      algorithms: Seq[H2OAlgorithm[_, _, _]]): Seq[BenchmarkBatch]  = {
+    def isAlgorithmBenchmark(clazz: Class[_]): Boolean = classOf[AlgorithmBenchmarkBase].isAssignableFrom(clazz)
```

Comment:
Algorithm Benchmark is different from the standard benchmark because it requires H2OContext?

Reply:
An algorithm benchmark instantiated and execute for each algorithm specified [here](https://github.com/h2oai/sparkling-water/pull/1345/files#diff-0627f7b73519122602ca58844dc8d658R73). Other benchmarks (like DF -> H2O frame conversions) are agnostic to algorithms.